mitodl/ocw-studio
# Modify RelationField to filter out unpublished content if the website property is present

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes #1602

#### What's this PR do?
Changes the behavior of `RelationField` so that if the `website` prop is specified, then only published content from that site will be available for selection.

#### How should this be manually tested?
- Add a new instructor to 'ocw-www', do not publish
- Go to any course site's metadata and try to add new instructors.  You should not see the new instructor, but should see and be able to select all others.
- Publish ocw-www to production.  Wait until it completes.
- Go back to the course site's metadata, you should now be able to add the new instructor.
- Add a small video resource to a course site, then create a video gallery and add that video.  You should be able to without publishing first.
- Go to the ocw-www site and create a new resource collection.  Make sure you can only add resources from sites that have been published to production.
